pylib/gyp/generator/msvs.py

```diff
@@ -3618,7 +3618,9 @@ def _AddSources2(
                     # Always add an <ObjectFileName> value to support duplicate
                     # source file basenames.
                     file_name = os.path.splitext(source)[0] + ".obj"
-                    if (file_name.startswith("..\\")):
+                    if (os.path.isabs(file_name)):
+                        file_name = os.path.splitdrive(file_name)[1]
+                    elif (file_name.startswith("..\\")):
                         file_name = re.sub(r"^(\.\.\\)+", "", file_name)
                     elif (file_name.startswith("$(")):
```

Comment:
OPTIONAL: Pythonic nit...
```suggestion
                    if os.path.isabs(file_name):
                        file_name = os.path.splitdrive(file_name)[1]
                    elif file_name.startswith("..\\"):
                        file_name = re.sub(r"^(\.\.\\)+", "", file_name)
                    elif file_name.startswith("$("):
```